### PR TITLE
Nexus HA IT plumbing

### PR DESCRIPTION
**Pull request by bentmann**
http://bamboo.s/browse/OSS-GOODIESF14-2

Imagine you run the HTTP test fixture and some docker container. "http://localhost:..." isn't going to do it for the docker container to talk to the HTTP fixture (living outside the container).

### CODE REVIEW

File: httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
 
   protected boolean ssl;
 
-  private final String host = "localhost"; // InetAddress.getLocalHost().getCanonicalHostName();
+  private String host = "localhost"; // InetAddress.getLocalHost().getCanonicalHostName();
```
Comment (sonatype-zion) [inline httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 16 issues:
* CRITICAL 5 critical
* MAJOR 10 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/api/ServerProvider.java#L89) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MINOR [Remove the declaration of thrown exception 'java.lang.IllegalStateException' which is a runtime exception.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/fluent/Server.java#L66) [rule](http://sonar.s/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Introduce a new variable instead of reusing the parameter "authName".](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L196) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1226)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L261) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L450) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L461) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L274) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L277) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L385) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Avoid calling toString() on String objects; this is unnecessary.](https://github.com/sonatype/goodies/blob/3f0a7a9be00acb9f99a87431f262a26dea782055/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L266) [rule](http://sonar.s/coding_rules#rule_key=pmd%3AStringToString)
* ... 5 more

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 15 issues:
* CRITICAL 5 critical
* MAJOR 9 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/api/ServerProvider.java#L89) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MINOR [Remove the declaration of thrown exception 'java.lang.IllegalStateException' which is a runtime exception.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/fluent/Server.java#L66) [rule](http://sonar.s/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Introduce a new variable instead of reusing the parameter "authName".](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L196) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1226)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L261) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L450) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* CRITICAL [Either log or rethrow this exception.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L461) [rule](http://sonar.s/coding_rules#rule_key=squid%3AS1166)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L274) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L277) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L385) [rule](http://sonar.s/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Avoid calling toString() on String objects; this is unnecessary.](https://github.com/sonatype/goodies/blob/a1d609f0c4b7aef2c6099df3eb86b59d1a54de22/httpfixture/src/main/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyServerProvider.java#L266) [rule](http://sonar.s/coding_rules#rule_key=pmd%3AStringToString)
* ... 5 more

**Comment by dbradicich** [comment]
not quite sure why sonar flipped out, but i dig it

+1

**Comment by cstamas** [comment]
+1

**Comment by kellyrob99** [comment]
+1

**Pull request merged**